src/utils/hooks/transaction/utils/extrinsic.ts

```diff
@@ -0,0 +1,127 @@
+    case Transaction.LOANS_LEND:
+      return window.bridge.loans.lend(...params.args);
+    case Transaction.LOANS_REPAY:
+      return window.bridge.loans.repay(...params.args);
+    case Transaction.LOANS_REPAY_ALL:
+      return window.bridge.loans.repayAll(...params.args);
+    case Transaction.LOANS_WITHDRAW:
+      return window.bridge.loans.withdraw(...params.args);
+    case Transaction.LOANS_WITHDRAW_ALL:
+      return window.bridge.loans.withdrawAll(...params.args);
+    case Transaction.LOANS_DISABLED_COLLATERAL:
+      return window.bridge.loans.disableAsCollateral(...params.args);
+    case Transaction.LOANS_ENABLED_COLLATERAL:
+      return window.bridge.loans.enableAsCollateral(...params.args);
+    /* END - LOANS */
+
+    /* START - LOANS */
+    case Transaction.VAULTS_DEPOSIT_COLLATERAL:
+      return window.bridge.vaults.depositCollateral(...params.args);
+    case Transaction.VAULTS_WITHDRAW_COLLATERAL:
+      return window.bridge.vaults.withdrawCollateral(...params.args);
+    case Transaction.VAULTS_REGISTER_NEW_COLLATERAL:
+      return window.bridge.vaults.registerNewCollateralVault(...params.args);
+    case Transaction.VAULT_WITHDRAW_REWARDS:
+      return window.bridge.rewards.withdrawRewards(...params.args);
+    /* END - LOANS */
+
+    /* START - ESCROW */
+    case Transaction.ESCROW_CREATE_LOCK:
+      return window.bridge.escrow.createLock(...params.args);
+    case Transaction.ESCROW_INCREASE_LOCKED_AMOUNT:
+      return window.bridge.escrow.increaseAmount(...params.args);
+    case Transaction.ESCROW_INCREASE_LOCKED_TIME:
+      return window.bridge.escrow.increaseUnlockHeight(...params.args);
+    case Transaction.ESCROW_WITHDRAW:
+      return window.bridge.escrow.withdraw(...params.args);
+    case Transaction.ESCROW_WITHDRAW_REWARDS:
+      return window.bridge.escrow.withdrawRewards(...params.args);
+    case Transaction.ESCROW_INCREASE_LOOKED_TIME_AND_AMOUNT: {
+      const [amount, unlockHeight] = params.args;
+      const txs = [
+        window.bridge.api.tx.escrow.increaseAmount(amount),
+        window.bridge.api.tx.escrow.increaseUnlockHeight(unlockHeight)
+      ];
+      const batch = window.bridge.api.tx.utility.batchAll(txs);
+
+      return { extrinsic: batch };
+    }
+    /* END - ESCROW */
+  }
+};
+
+/**
+ * The status where we want to be notified on the transaction completion
+ * @param {Transaction} type type of transaction
+ * @return {ExtrinsicStatus.type} transaction status
+ */
+const getExpectedStatus = (type: Transaction): ExtrinsicStatus['type'] => {
+  switch (type) {
+    // When requesting a replace, wait for the finalized event because we cannot revert BTC transactions.
+    // For more details see: https://github.com/interlay/interbtc-api/pull/373#issuecomment-1058949000
+    case Transaction.ISSUE_REQUEST:
+    case Transaction.REDEEM_REQUEST:
+    case Transaction.REPLACE_REQUEST:
+      return 'Finalized';
+    default:
+      return 'InBlock';
+  }
+};
```

Comment:
Just a questions here: is there any way how we can specify the status in `useTransaction` call? In case there are situations where the same lib call needs to be treated differently (e.g. in one part of dapp we wait for finality, but somewhere else inBlock inclusion might be sufficient. Or will we need to specify another case in Transaction enum in such case?

Reply:
Excelent idea.